Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -569,6 +569,220 @@ class UpstreamIntegrationTests: TestBase {
+
+        XCTAssertEqual(1, matchingResponses.count)
+        XCTAssertEqual(400, matchingResponses.first?.responseCode)
+    }
+
+    // Tests datastream config overrides
+    func testSendEvent_withInvalidConfigOverrides_invalidConfigValues_receivesExpectedNetworkResponseError() {
+        // Setup
+        // Note: test constructs should always be valid
+        let interactNetworkRequest = NetworkRequest(urlString: createURLWith(locationHint: edgeLocationHint), httpMethod: .post)!
+        // Setting expectation allows for both:
+        // 1. Validation that the network request was sent out
+        // 2. Waiting on a response for the specific network request (with timeout)
+        networkService.setExpectationForNetworkRequest(networkRequest: interactNetworkRequest, expectedCount: 1)
+        // Overriding with dataset not added in the datastream config
+        // And Overriding with RSID not added to override setting in the Analytics upstream config
+        let configOverrides = ["com_adobe_experience_platform": [
+                                    "datasets": [
+                                        "event": [
+                                            "datasetId": "do_not_delete_dataset3_e2e_testing_edge_konductor"
+                                        ]
+
+                                    ]
+                                ],
+                                "com_adobe_analytics": [
+                                    "reportSuites": [
+                                        "mobile5e2e.rsid3"
+                                    ]
+                                ]]
+        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"], data: ["data": ["test": "data"]], datastreamConfigOverride: configOverrides)
+
+        // Test
+        Edge.sendEvent(experienceEvent: experienceEvent)
+
+        // Verify
+        // Network response assertions
+        let matchingResponses = networkService.getResponsesFor(networkRequest: interactNetworkRequest, timeout: 5)
+
+        XCTAssertEqual(1, matchingResponses.count)
+        XCTAssertEqual(400, matchingResponses.first?.responseCode)
+    }
+
+    func testSendEvent_withInvalidConfigOverrides_invalidDummyValues_receivesExpectedNetworkResponseError() {
+        // Setup
+        // Note: test constructs should always be valid
+        let interactNetworkRequest = NetworkRequest(urlString: createURLWith(locationHint: edgeLocationHint), httpMethod: .post)!
+        // Setting expectation allows for both:
+        // 1. Validation that the network request was sent out
+        // 2. Waiting on a response for the specific network request (with timeout)
+        networkService.setExpectationForNetworkRequest(networkRequest: interactNetworkRequest, expectedCount: 1)
+        // Overriding with dataset not added in the datastream config
+        // And Overriding with RSID not added to override setting in the Analytics upstream config
```

Comment:
nit: I think you can move this as a description of the test for improved readability - same for all usages